### PR TITLE
Apply garbage collection for tree

### PR DESCRIPTION
**Pull request by JOOHOJANG**
**What this PR does / why we need it**:
- Apply garbage collection for tree
- Fix some issues
  - miscalculation in `pathToTreePos`
  - null point exception when `Node.Parent` is `nil`

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: pkg/document/crdt/tree.go [hunk 1]
```diff
@@ -433,7 +498,7 @@ func (t *Tree) Edit(from, to *TreePos, content *TreeNode, editedAt *time.Ticket)
 				}
 			}
 		} else {
-			if fromPos.Node.Parent.Value.IsRemoved() {
+			if fromPos.Node.Parent != nil && fromPos.Node.Parent.Value.IsRemoved() {
```
Comment (hackerwins) [inline pkg/document/crdt/tree.go:501]: May I know in which case the problem occurs?

I guess that `fromPos.node` is the root node, I'm wondering if we can just ignore it or if we need to do something else.

**Review by hackerwins — COMMENTED** [review]
Thanks for your contribution.
I left a simple question.

File: pkg/document/crdt/tree.go (hunk 1 above)
Comment (JOOHOJANG) [reply pkg/document/crdt/tree.go:501]: As you said, we need this line of code when fromPos.Node is root of the tree. 
I think no other process is needed

File: pkg/document/crdt/tree.go [hunk 2]
```diff
@@ -282,6 +285,64 @@ func (t *Tree) Marshal() string {
 	return builder.String()
 }
 
+// returns the length of removed nodes
```
Comment (hackerwins) [inline pkg/document/crdt/tree.go:288]:
```suggestion
// removedNodesLen returns the length of removed nodes.
```

File: test/integration/gc_test.go [hunk 3]
```diff
@@ -20,6 +20,7 @@ package integration
 
 import (
 	"context"
+	"github.com/yorkie-team/yorkie/pkg/document/time"
```
Comment (hackerwins) [inline test/integration/gc_test.go:23]: Could you put this import to the below group?

https://github.com/uber-go/guide/blob/master/style.md#import-group-ordering

File: pkg/index/tree.go [hunk 4]
```diff
@@ -93,6 +93,18 @@ func postOrderTraversal[V Value](node *Node[V], callback func(node *Node[V], dep
 	callback(node, depth)
 }
 
+// postOrderTraversalAll traverses the whole tree (include tombstones) with postorder traversal.
+func postOrderTraversalAll[V Value](node *Node[V], callback func(node *Node[V], depth int), depth int) {
```
Comment (hackerwins) [inline pkg/index/tree.go:97]:
```suggestion
func postorderTraversalAll[V Value](node *Node[V], callback func(node *Node[V], depth int), depth int) {
```

Previous codes also did, but I think we could replace `postOrder` with `postorder`.

**Review by hackerwins — APPROVED** [review]
Thanks for your reply.

I only left minor style comments.

**Pull request merged**